target/linux/ar71xx/files/arch/mips/ath79/mach-fritz1750e.c

```diff
@@ -0,0 +1,252 @@
+/*
+ *  AVM FRITZ!WLAN Repeater 1750E board support
+ *
+ *  Copyright (C) 2018 David Bauer <mail@david-bauer.net>
+ *
+ *  This program is free software; you can redistribute it and/or modify it
+ *  under the terms of the GNU General Public License version 2 as published
+ *  by the Free Software Foundation.
+ */
+
+#include <linux/init.h>
+#include <linux/ath9k_platform.h>
+#include <linux/etherdevice.h>
+#include <linux/gpio.h>
+#include <linux/platform_device.h>
+
+#include <linux/mtd/mtd.h>
+#include <linux/mtd/partitions.h>
+#include <linux/platform_data/mdio-gpio.h>
+#include <linux/platform_data/phy-at803x.h>
+#include <linux/spi/spi_gpio.h>
+#include <linux/spi/74x164.h>
+
+#include <asm/mach-ath79/ath79.h>
+#include <asm/mach-ath79/ar71xx_regs.h>
+
+#include "common.h"
+#include "dev-eth.h"
+#include "dev-gpio-buttons.h"
+#include "dev-leds-gpio.h"
+#include "dev-m25p80.h"
+#include "dev-spi.h"
+#include "dev-wmac.h"
+#include "machtypes.h"
+
+
+#define FRITZ1750E_GPIO_SHIFT_SER	15   /* DS,   Data Serial Input */
+#define FRITZ1750E_GPIO_SHIFT_SRCLK	14 /* SHCP, Shift Reg Clock Input */
+
+#define FRITZ1750E_SSR_BIT_0			0
+#define FRITZ1750E_SSR_BIT_1			1
+#define FRITZ1750E_SSR_BIT_2			2
+#define FRITZ1750E_SSR_BIT_3			3
+#define FRITZ1750E_SSR_BIT_4			4
+#define FRITZ1750E_SSR_BIT_5			5
+#define FRITZ1750E_SSR_BIT_6			6
+#define FRITZ1750E_SSR_BIT_7			7
+
+#define FRITZ1750E_74HC_GPIO_BASE		32
+#define FRITZ1750E_74HC_GPIO_LED_RSSI0	(FRITZ1750E_74HC_GPIO_BASE + 0)
+#define FRITZ1750E_74HC_GPIO_LED_RSSI1	(FRITZ1750E_74HC_GPIO_BASE + 1)
+#define FRITZ1750E_74HC_GPIO_LED_RSSI2		(FRITZ1750E_74HC_GPIO_BASE + 2)
+#define FRITZ1750E_74HC_GPIO_LED_RSSI3		(FRITZ1750E_74HC_GPIO_BASE + 3)
+#define FRITZ1750E_74HC_GPIO_LED_RSSI4		(FRITZ1750E_74HC_GPIO_BASE + 4)
+#define FRITZ1750E_74HC_GPIO_LED_WLAN		  (FRITZ1750E_74HC_GPIO_BASE + 5)
+#define FRITZ1750E_74HC_GPIO_LED_POWER		  (FRITZ1750E_74HC_GPIO_BASE + 6)
+
+#define FRITZ1750E_GPIO_LED_LAN	13
+
+#define FRITZ1750E_GPIO_BTN_WPS	4
+#define FRITZ1750E_KEYS_POLL_INTERVAL	 	20 /* msecs */
+#define FRITZ1750E_KEYS_DEBOUNCE_INTERVAL	(3 * FRITZ1750E_KEYS_POLL_INTERVAL)
+
+#define FRITZ1750E_PHY_MDIO_ID 0
+#define FRITZ1750E_GPIO_PHY_RESET 11
+#define FRITZ1750E_GPIO_MDIO_CLK 12
+#define FRITZ1750E_GPIO_MDIO_DATA 19
+
+#define FRTIZ1750E_OFFSET_TFFS1_LAN_MAC 0x58C
+
+
+static struct mdio_gpio_platform_data fritz1750e_mdio_data = {
+	.mdc = FRITZ1750E_GPIO_MDIO_CLK,
+	.mdio = FRITZ1750E_GPIO_MDIO_DATA,
+	.phy_mask = ~BIT(FRITZ1750E_PHY_MDIO_ID),
+};
+
+static struct at803x_platform_data fritz1750e_ar8033_data = {
+	.has_reset_gpio = 1,
+	.reset_gpio = FRITZ1750E_GPIO_PHY_RESET,
+};
+
+static struct platform_device fritz1750e_phy_device = {
+	.name = "mdio-gpio",
+	.id = 0,
+	.dev = {
+		.platform_data = &fritz1750e_mdio_data, &fritz1750e_ar8033_data
```

Comment:
this cant be right.

Reply:
Honestly, i've just oriented myself how it was handled with the EAP120 ( https://github.com/openwrt/openwrt/blob/master/target/linux/ar71xx/files/arch/mips/ath79/mach-eap120.c#L84 ). Do you have an idea how to achieve the goal of using the mdio_gpio_platform_data and at803x_platform_data?